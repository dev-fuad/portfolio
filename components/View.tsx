import { View as RNView, ViewProps as RNViewProps } from 'react-native';

import { getStyleProps } from '@/theme/styles';

export interface ViewProps extends RNViewProps {
  className?: string;
}

export default function View({ className, ...props }: ViewProps) {
  return <RNView {...props} {...getStyleProps(className, props.style)} />;
}
