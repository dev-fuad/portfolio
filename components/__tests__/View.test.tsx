import { render } from '@testing-library/react-native';

import View from '@/components/View';

describe('View component', () => {
  it('renders correctly with default props', () => {
    const { toJSON } = render(<View />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies className styles correctly', () => {
    const { getByTestId } = render(<View className="test-class" testID="view" />);
    const view = getByTestId('view');
    expect(view.props.style).toBeDefined();
  });

  it('passes other props to RNView', () => {
    const { getByTestId } = render(<View testID="view" accessibilityLabel="test-label" />);
    const view = getByTestId('view');
    expect(view.props.accessibilityLabel).toBe('test-label');
  });
});