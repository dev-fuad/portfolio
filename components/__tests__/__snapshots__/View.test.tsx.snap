// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View component renders correctly with default props 1`] = `<View />`;
